### DISCUSSION TITLE
Safari tabs and windows

### DISCUSSION THREAD

**Discussion by aalennard** [ttscoff/bunch-docs#145] [Q&A] [answered]
Is there way to consistently get Safari to open specific websites as tabs in specific new windows? And to ensure sequential processing?

I've tried the example below (and many other iterations) that end up with all 4 tabs on one Safari window, and THEN, at the end, Safari opens two additional windows. It seems like Bunch is not processing sequentially, but I've tried adding "sequential: true" or "sequence: sequential" to no effect, and the global setting for parallel processing is disabled.

%Safari
- {@n !3}
- https://backblaze.com
- http://cisco.com
- {@n !3}
- https://www.wsj.com
- https://latimes.com

If I add pauses in a different way, it partially works. The following opens the 1st and 2nd URLS open as tabs in any existing Safari window, then does open the 3rd and 4th URLS as tabs in their own window, and then a 3rd safari window opens to the start page).

%Safari
- {@n}
- (pause 3)
- https://backblaze.com
- http://cisco.com
- {@n}
- (pause 3)
- https://www.wsj.com
- https://latimes.com

MacOS 11.6
Safari 15.0

Thanks!

**Comment by ttscoff** [comment, marked as answer]
This is a duplicate of #60, which you can check for more info.

Timing gets weird when sending both keystrokes and urls/files to an app at the same time. Sequential execution does well with app launching and command execution, but not as well with opening files/urls because there's no way for Bunch to know when an action is completed, how long it took an app to respond to the open command, etc.

The workaround is hacky, but I've tested it and it works pretty flawlessly (faster than keystroke pauses, too):

```
%Safari
- {@n "https://backblaze.com" return @t "http://cisco.com" return}
- {@n "https://www.wsj.com" return @t "https://latimes.com" return}
```

**Comment by aalennard** [comment]
This works great. Thank you.
